### DISCUSSION TITLE
blog/2021/12/18/bayesian-propensity-scores-weights/index

### DISCUSSION THREAD

**Discussion by giscus** [andrewheiss/ath-quarto#65] [Blog comments]
# blog/2021/12/18/bayesian-propensity-scores-weights/index

For mathematical and philosophical reasons, propensity scores and inverse probability weights don’t work in Bayesian inference. But never fear! There’s still a way to do it!

https://www.andrewheiss.com/blog/2021/12/18/bayesian-propensity-scores-weights/index.html

**Comment by e-pet** [comment]
Just to make sure I understand this correctly... if I do 
1) estimate IPTW using a treatment regression model,
2) sample a (large) re-weighted pseudopopulation from the original data using those weights, and
3) perform Bayesian inference on that pseudopopulation (for other factors than the ones I adjusted for above)

... what I am missing / doing "wrong" is that I am neglecting the uncertainty afflicting the IPTWs?

**Comment by LukeLB** [comment]
This and the companion blog were a great read Andrew, thank your for writing them. I'm currently working through Matheus Facure's book (https://www.amazon.co.uk/Causal-Inference-Python-Applying-Industry/dp/1098140257) and as an exercise translating it to PyMC/Bambi/CausalPy. In the book the ATE is calculated directly without an outcome model using the equation, \
 $ATE=E[Y\frac{T-e(X)}{e(X)(1-e(X))}]$ \
where, Y is the outcome, T is the treatment, and e(X) is the propensity score. In this case it is just a transformation of the posterior propensity scores which I think gives you the posterior ATE. This appears to avoid the problem of having to fit multiple frequentist outcome models which as you say leaves some uncertainty on the floor, but I'm not sure if I'm missing something...

**Comment by jrgant** [comment]
Great post on something I've been trying to wrap my head around!

One note: frequentist inference shouldn't be conflated with NHST. One can simply focus on estimating the ATE and computing a confidence interval. You can derive a p-value from the interval and test if you want (I personally wouldn't), but the focus rests simply on quantifying the uncertainty due to random error. Of course, this isn't the posterior inference you're looking for, but the approach, while frequentist, does not adopt the philosophy of NHST.

**Comment by tinghua-chen** [comment]
Thanks for your post—this is incredibly helpful! I've been reading about how to incorporate propensity score models into Bayesian statistics. I’ve noticed that some researchers use posterior predictive inference methods. Specifically, they obtain posterior draws from the propensity score model and the outcome model based on their respective posterior predictive distributions, and then plug these posterior draws into doubly robust estimators.

However,  I’m having difficulty understanding the benefits of using this approach compared to the two-stage method you discuss here. Do you have any thoughts or insights on this? For reference, here’s a related paper: https://onlinelibrary.wiley.com/doi/full/10.1111/biom.13417 and https://academic.oup.com/biomet/article/103/3/667/1743939
